### PR TITLE
ci(docs): Check broken links

### PR DESCRIPTION
**Pull request by zharinov**
## Changes

Leverage [lychee CLI tool](https://github.com/lycheeverse/lychee) to check internal and external links in the documentation.

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by rarkins — DISMISSED** [review]

File: lib/modules/manager/tekton/readme.md
Comment (HonkingGoose) [inline lib/modules/manager/tekton/readme.md:20]:
```suggestion
By specifying the annotation with a remote task or a remote pipeline based on the recommended way using [git based versioning](https://github.com/tektoncd/community/blob/main/teps/0115-tekton-catalog-git-based-versioning.md).
How this can be used can be seen in the example below.
```

Let's make this one sentence per line.

File: .github/lychee.toml
Comment (HonkingGoose) [inline .github/lychee.toml:16]: According to the docs:

> `  -i, --insecure
          Proceed for server connections considered insecure (invalid TLS)
`

Do we need this flag?

**Review by HonkingGoose — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml
Comment (viceice) [inline .github/workflows/build.yml:600]: use list style

File: docs/usage/examples/opentelemetry.md
Comment (viceice) [inline docs/usage/examples/opentelemetry.md:133]: do we really want to pin this?

File: lib/modules/versioning/ruby/index.ts
Comment (viceice) [inline lib/modules/versioning/ruby/index.ts:22]: why removed?

File: .github/workflows/build.yml
Comment (viceice) [inline .github/workflows/build.yml:635]: this should not accidentally fail if external urls change. will be really bad ci experience

File: docs/usage/examples/opentelemetry.md
Comment (zharinov) [reply docs/usage/examples/opentelemetry.md:133]: The current link is broken, `spanmetricprocessor` was deprecated and removed

File: .github/workflows/build.yml
Comment (zharinov) [reply .github/workflows/build.yml:634]: I agree, it should be annoying enough for us to fix links ASAP, but shouldn't prevent us from :shipit:

File: lib/modules/versioning/ruby/index.ts
Comment (rarkins) [reply lib/modules/versioning/ruby/index.ts:22]: This link keeps changing, let's leave it off

**Comment by zharinov** [comment]
Now it will fail, but this won't block merge because none of required tasks depend on it

**Review by viceice — DISMISSED** [review]

File: lib/modules/versioning/ruby/index.ts
Comment (zharinov) [inline lib/modules/versioning/ruby/index.ts:21]:
```suggestion
  'https://guides.rubygems.org/patterns/',
  'https://bundler.io/v2.5/man/gemfile.5.html',
```

File: .github/lychee.toml
Comment (HonkingGoose) [inline .github/lychee.toml:17]: The [lychee docs, rate limiting](https://lychee.cli.rs/troubleshooting/rate-limits/) list steps to reduce the number of 429 responses. Accepting `429` rate limited responses is the last resort. So maybe we should try the other things the docs list?

| Method | Do we use it? | Notes |
| :---- | :--- | :--- | 
| Do not retry a failed link, with `--max-retries 0` | No | |
| Set lower `--max-concurrency` | No | Lower concurrency makes the action slower! |
| Use `GITHUB_TOKEN` | No | A personal token with no extra permissions will work to check public links. |
| Caching | Yes, 1 day | |
| Ignore certain sites or paths | Yes | | 
| Accept `429` responses | Yes | Last resort according to the lychee docs. |

File: .github/workflows/build.yml
Comment (HonkingGoose) [inline .github/workflows/build.yml:603]: The [GitHub Docs, `timeout-minutes`](https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#jobsjob_idtimeout-minutes) says:

> The maximum number of minutes to let a job run before GitHub automatically cancels it. Default: 360

> **Note:** The `GITHUB_TOKEN` expires when a job finishes or after a maximum of 24 hours.

Are you sure that 2 minutes is enough for the action to run succesfully? This timeout seems very low to me.

File: .github/lychee.toml
Comment (HonkingGoose) [inline .github/lychee.toml:1]: I found these security related issues. I'll let you review them:

- https://github.com/lycheeverse/lychee/issues/1297
- https://github.com/lycheeverse/lychee/issues/1298

File: .github/lychee.toml
Comment (zharinov) [reply .github/lychee.toml:17]: We have an additional step to remove `429` responses from the cache, so the next run they will be retried. Eventually, all the rate limited links will be       checked (during 24 hours).

File: .github/workflows/build.yml
Comment (zharinov) [reply .github/workflows/build.yml:603]: It's okay, in my laptop runs for about 10 seconds

File: .github/lychee.toml
Comment (HonkingGoose) [reply .github/lychee.toml:17]: #### My summary of your comment

Here's how I understood your comment:

- We get lots of PRs per day. Either from Renovate bot, or us, or contributors.
- This means the link check runs at least once for each incoming PR.
- Because the link check runs often: we can swallow the `429` and "hope" that on the next run(s) the link checker gets a good response.
- Over the course of a day, we checked all rate-limited links.

#### One day cache long enough?

Is a day enough time to be reasonably sure we checked all the links? I mean, is it worth it to have a longer time to live for the cache? Say 2 days?

Comment (zharinov) [reply .github/lychee.toml:17]: 1 day is ok

File: .github/lychee.toml
Comment (zharinov) [reply .github/lychee.toml:1]: 🤷‍♂️

**Comment by HonkingGoose** [comment]
We just merged this PR, that makes it so we use Material for MkDocs to build our docs. The publish step stays in the Renovate docs repository!

- #30735

I asked @viceice what we want to do with your PR:

> > What do you want to do with this PR?
> > 
> > * [ci(docs): Check broken links #30285](https://github.com/renovatebot/renovate/pull/30285)
> 
> I think it's obsolete then 🤷‍♂️

Because we're building with Material for MkDocs here now, we probably can rely more on their built-in link checks.

We should test if the docs build on `renovatebot/renovate` works as we expect. And then decide if we want to use the Lychee stuff.

In any case, if we decide to try Lychee, you should merge in the latest changes from `main` to make sure we're not having any logical conflicts with the docs build on `main`. 😉

**Pull request closed without merging**
